crates/flux-middle/src/def_id.rs

```diff
@@ -22,18 +22,36 @@ use rustc_span::Symbol;
 ///   when item existence is guaranteed
 /// * The type is parametric over the parent `Id` type to support various id types (e.g., [`DefId`],
 ///   [`MaybeExternId`])
-#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq, Encodable, Decodable)]
+#[derive(Debug, Copy, Clone, Encodable, Decodable)]
 pub struct FluxId<Id> {
     parent: Id,
     name: Symbol,
+    span: Span,
```

Comment:
I really don't like this. Please create a query `flux_def_ident_span` (similar to `TyCtxt::def_ident_span`) and query the span when you need it.